From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Sat, 4 Oct 2025 23:26:42 +0530
Subject: [PATCH] trace: Make code compatible for older assign_str arguments

After changing assign_str to one argument in KV6.10, backporting to
older version need to change parameters with in trace definition.
Go back to old handling.

Error:
--------------------------------------------------------------------------
error: macro "__assign_str" requires 2 arguments, but only 1 given
--------------------------------------------------------------------------

Reference-id: 2c92ca849fcc tracing/treewide: Remove second parameter of
__assign_str()

Signed-off-by: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/gpu/drm/scheduler/gpu_scheduler_trace.h   | 10 ++++++-
 drivers/gpu/drm/xe/xe_trace.h             | 28 +++++++++++++++++++
 drivers/gpu/drm/xe/xe_trace_bo.h          | 21 ++++++++++++++
 drivers/gpu/drm/xe/xe_trace_guc.h         | 13 +++++++++
 drivers/gpu/drm/xe/xe_trace_lrc.h         |  5 ++++
 5 files changed, 76 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/scheduler/gpu_scheduler_trace.h b/drivers/gpu/drm/scheduler/gpu_scheduler_trace.h
index f56e77e..2134f84 100644
--- a/drivers/gpu/drm/scheduler/gpu_scheduler_trace.h
+++ b/drivers/gpu/drm/scheduler/gpu_scheduler_trace.h
@@ -48,7 +48,11 @@ DECLARE_EVENT_CLASS(drm_sched_job,
 			   __entry->entity = entity;
 			   __entry->id = sched_job->id;
 			   __entry->fence = &sched_job->s_fence->finished;
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(name, sched_job->sched->name);
+#else			   
 			   __assign_str(name);
+#endif			   
 			   __entry->job_count = spsc_queue_count(&entity->job_queue);
 			   __entry->hw_job_count = atomic_read(
 				   &sched_job->sched->credit_count);
@@ -94,7 +98,11 @@ TRACE_EVENT(drm_sched_job_wait_dep,
 			     ),
 
 	    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+   			   __assign_str(name, sched_job->sched->name);
+#else	    
 			   __assign_str(name);
+#endif			  
 			   __entry->id = sched_job->id;
 			   __entry->fence = fence;
 			   __entry->ctx = fence->context;
@@ -110,5 +118,5 @@ TRACE_EVENT(drm_sched_job_wait_dep,
 
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm/scheduler
+#define TRACE_INCLUDE_PATH ../drivers/gpu/drm/scheduler
 #include <trace/define_trace.h>
diff --git a/drivers/gpu/drm/xe/xe_trace.h b/drivers/gpu/drm/xe/xe_trace.h
index d5281de..7a3a721 100644
--- a/drivers/gpu/drm/xe/xe_trace.h
+++ b/drivers/gpu/drm/xe/xe_trace.h
@@ -36,7 +36,11 @@ DECLARE_EVENT_CLASS(xe_gt_tlb_invalidation_fence,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_xe(xe));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->fence = fence;
 			   __entry->seqno = fence->seqno;
 			   ),
@@ -97,7 +101,11 @@ DECLARE_EVENT_CLASS(xe_exec_queue,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_eq(q));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->class = q->class;
 			   __entry->logical_mask = q->logical_mask;
 			   __entry->gt_id = q->gt->info.id;
@@ -221,7 +229,11 @@ DECLARE_EVENT_CLASS(xe_sched_job,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_eq(job->q));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->seqno = xe_sched_job_seqno(job);
 			   __entry->lrc_seqno = xe_sched_job_lrc_seqno(job);
 			   __entry->gt_id = job->q->gt->info.id;
@@ -288,7 +300,11 @@ DECLARE_EVENT_CLASS(xe_sched_msg,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_eq(((struct xe_exec_queue *)msg->private_data)));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->opcode = msg->opcode;
 			   __entry->guc_id =
 			   ((struct xe_exec_queue *)msg->private_data)->guc->id;
@@ -322,7 +338,11 @@ DECLARE_EVENT_CLASS(xe_hw_fence,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_xe(fence->xe));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->ctx = fence->dma.context;
 			   __entry->seqno = fence->dma.seqno;
 			   __entry->fence = fence;
@@ -361,7 +381,11 @@ TRACE_EVENT(xe_reg_rw,
 		),
 
 	TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		__assign_str(dev, __dev_name_tile(mmio->tile));
+#else
 		__assign_str(dev);
+#endif
 		__entry->val = val;
 		__entry->reg = reg;
 		__entry->write = write;
@@ -385,7 +409,11 @@ DECLARE_EVENT_CLASS(xe_pm_runtime,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_xe(xe));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->caller = caller;
 			   ),
 
diff --git a/drivers/gpu/drm/xe/xe_trace_bo.h b/drivers/gpu/drm/xe/xe_trace_bo.h
index ea50fee..f8e25d9 100644
--- a/drivers/gpu/drm/xe/xe_trace_bo.h
+++ b/drivers/gpu/drm/xe/xe_trace_bo.h
@@ -32,7 +32,11 @@ DECLARE_EVENT_CLASS(xe_bo,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_bo(bo));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->size = bo->size;
 			   __entry->flags = bo->flags;
 			   __entry->vm = bo->vm;
@@ -69,9 +73,18 @@ TRACE_EVENT(xe_bo_move,
 	    TP_fast_assign(
 		   __entry->bo      = bo;
 		   __entry->size = bo->size;
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		   __assign_str(new_placement_name, xe_mem_type_to_name[new_placement]);
+		   __assign_str(old_placement_name, xe_mem_type_to_name[old_placement]);
+#else
 		   __assign_str(new_placement_name);
 		   __assign_str(old_placement_name);
+#endif
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		   __assign_str(device_id,  __dev_name_bo(bo));
+#else
 		   __assign_str(device_id);
+#endif
 		   __entry->move_lacks_source = move_lacks_source;
 		   ),
 	    TP_printk("move_lacks_source:%s, migrate object %p [size %zu] from %s to %s device_id:%s",
@@ -94,7 +107,11 @@ DECLARE_EVENT_CLASS(xe_vma,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_vma(vma));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->vma = vma;
 			   __entry->asid = xe_vma_vm(vma)->usm.asid;
 			   __entry->start = xe_vma_start(vma);
@@ -188,7 +205,11 @@ DECLARE_EVENT_CLASS(xe_vm,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_vm(vm));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->vm = vm;
 			   __entry->asid = vm->usm.asid;
 			   ),
diff --git a/drivers/gpu/drm/xe/xe_trace_guc.h b/drivers/gpu/drm/xe/xe_trace_guc.h
index 78949db..2a97c31 100644
--- a/drivers/gpu/drm/xe/xe_trace_guc.h
+++ b/drivers/gpu/drm/xe/xe_trace_guc.h
@@ -32,7 +32,11 @@ DECLARE_EVENT_CLASS(xe_guc_ct_flow_control,
 			     ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			   __assign_str(dev, __dev_name_xe(xe));
+#else
 			   __assign_str(dev);
+#endif
 			   __entry->_head = _head;
 			   __entry->_tail = _tail;
 			   __entry->size = size;
@@ -73,7 +77,11 @@ DECLARE_EVENT_CLASS(xe_guc_ctb,
 		    ),
 
 		    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			    __assign_str(dev, __dev_name_xe(xe));
+#else
 			    __assign_str(dev);
+#endif
 			    __entry->gt_id = gt_id;
 			    __entry->action = action;
 			    __entry->len = len;
@@ -124,8 +132,13 @@ TRACE_EVENT(xe_guc_engine_activity,
 	    ),
 
 	    TP_fast_assign(
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+			__assign_str(dev, __dev_name_xe(xe));
+    			__assign_str(name, name);
+#else			
 			__assign_str(dev);
 			__assign_str(name);
+#endif			
 			__entry->global_change_num = ea->metadata.global_change_num;
 			__entry->guc_tsc_frequency_hz = ea->metadata.guc_tsc_frequency_hz;
 			__entry->lag_latency_usec = ea->metadata.lag_latency_usec;
diff --git a/drivers/gpu/drm/xe/xe_trace_lrc.h b/drivers/gpu/drm/xe/xe_trace_lrc.h
index d525cbe..4653bd5 100644
--- a/drivers/gpu/drm/xe/xe_trace_lrc.h
+++ b/drivers/gpu/drm/xe/xe_trace_lrc.h
@@ -33,8 +33,13 @@ TRACE_EVENT(xe_lrc_update_timestamp,
 		   __entry->lrc	= lrc;
 		   __entry->old = old;
 		   __entry->new = lrc->ctx_timestamp;
+#ifdef BPM_ASSIGN_STR_SECOND_ARG_PRESENT
+		   __assign_str(name, lrc->fence_ctx.name);
+		   __assign_str(device_id,  __dev_name_lrc(lrc));
+#else
 		   __assign_str(name);
 		   __assign_str(device_id);
+#endif
 		   ),
 	    TP_printk("lrc=:%p lrc->name=%s old=%llu new=%llu device_id:%s",
 		      __entry->lrc, __get_str(name),
-- 
2.43.0

